addons/membership/models/partner.py

```diff
@@ -86,12 +86,9 @@ def _compute_membership_stop(self):
                  'associate_member.membership_state')
     def _compute_membership_cancel(self):
         for partner in self:
-            if partner.membership_state == 'canceled':
-                partner.membership_cancel = self.env['membership.membership_line'].search([
-                    ('partner', '=', partner.id)
-                ], limit=1, order='date_cancel').date_cancel
-            else:
-                partner.membership_cancel = False
+            partner.membership_cancel = self.env['membership.membership_line'].search([
+                ('partner', '=', partner.id)
```

Comment:
While I am trying to understand all the code of the this file, I have a question. Why this line can't be
```python
('partner', '=', partner.associate_member.id or partner.id),('date_cancel','=',False)
```
?

Reply:
maybe add a domain on the company (in case of multi company).